tests/integration/extenders/NotificationTest.php

```diff
@@ -119,23 +125,64 @@ public function notification_driver_enabled_types_exist_if_added()
         $this->assertContains('secondCustomDriver', $blueprints[SecondCustomNotificationType::class]);
         $this->assertEmpty($blueprints[ThirdCustomNotificationType::class]);
     }
+
+    /**
+     * @test
+     */
+    public function notification_before_sending_callback_works_if_added()
+    {
+        $this->extend(
+            (new Extend\Notification)
+                ->type(CustomNotificationType::class, 'customNotificationTypeSerializer')
+                ->driver('customNotificationDriver', CustomNotificationDriver::class)
+                ->beforeSending(function ($blueprint, $users) {
+                    if ($blueprint instanceof CustomNotificationType) {
+                        unset($users[1]);
+                    }
+
+                    return $users;
+                })
+        );
+
+        $this->prepareDatabase([
+            'discussions' => [
+                ['id' => 1, 'title' => __CLASS__, 'created_at' => Carbon::now()->toDateTimeString(), 'user_id' => 3, 'first_post_id' => 1, 'comment_count' => 1],
```

Comment:
Is this used somewhere in the test case?

Reply:
Nvm, found it below. I'm still not sure it's necessary though, see the commend below.